### DISCUSSION TITLE
Possible bug when combining Annotated and a Form with an alias preventing generating documentation Annotated[str, Form(alias="some-alias")]

### DISCUSSION THREAD

**Discussion by ashkan117** [fastapi/fastapi#9877] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Working variations
working_form1 = Form(alias="some-alias")
working_form2 = Annotated[str, Form()]
# Below fails to work
broken_form = Annotated[str, Form(alias="some-alias")]
```

### Description

1. Create an endpoint that has an Annotated parameter that Depends on another Annotated parameter that is a Form with a specified alias. 
2. Generate the docs by going to localhost:8000/docs 
3. An error is thrown saying 
```bash
TypeError: Object of type '_AnnotatedAlias' is not JSON serializable
```

After taking a look at a few things I think it's tied to Pydantic but not sure how the interaction between Pydantic and FastAPI work since Annotated Forms with no alias work fine.

From what I understand the difference is that an Annotated Form with an alias has the type (type[str]) whereas the one with no alias has the type (str). This difference causes the following error in Pydantic
```
    # pydantic_encoder function 
    # https://github.com/pydantic/pydantic/blob/e75ffbd978aff64a57581e5c52e00f520f04fa40/pydantic/v1/json.py#L83
    for base in obj.__class__.__mro__[:-1]:
```

It looks like type[str] would need to be using the obj.mro() instead?

### Operating System

Linux

### FastAPI Version

0.98.0

### Python Version

3.11.3

**Comment by Kludex** [comment]
Please provide a code I can run.

**Reply by hochstibe** [reply]
I could not reproduce in with 0.98 or 0.100 (pydantic v1)

With v0.100 and pydantic v2, the alias is not respected in openapi (as in #9869). Because the post parameter is required, an `Unprocessable Entity` is raised.

```python
from typing import Annotated

from fastapi import FastAPI, Form, Query


app = FastAPI()


@app.post("/")
def alias(
    # schema_: Annotated[str, Form(alias="some-alias")] | None = None,
    schema_: Annotated[str, Form(alias="some-alias")],  # --> unprocessable entity
):
    return {"schema": schema_}
```

**Reply by ashkan117** [reply]
I am working on providing an environment where people can run. In doing so I noticed a difference between the project I am working on and the sample project I am building. In my project I notice that the type hint shows that it returns type[str] but in my test project it's just str.

Is there a reason why one is type[str] and the other is str

![type str -fastapi](https://github.com/tiangolo/fastapi/assets/30095041/ab76d855-f593-423c-bcf0-d08a4a467afb)

whereas in the new project its type is just str

![image](https://github.com/tiangolo/fastapi/assets/30095041/0b3ffaed-2fc1-4c0b-ad56-2a51031f083b)

**Comment by Kludex** [comment, marked as answer]
You can track this issue on https://github.com/tiangolo/fastapi/issues/10286.
